packages/orchestrator-ui-components/src/components/WfoSubscription/WfoSubscriptionProductBlock/WfoSubscriptionProductBlock.tsx

```diff
@@ -159,19 +169,22 @@ export const WfoSubscriptionProductBlock = ({
                                 </>
                             )}
 
-                            {productBlockInstanceValues
-                                .filter(
-                                    (productBlockInstanceValue) =>
-                                        !HIDDEN_KEYS.includes(
-                                            productBlockInstanceValue.field,
-                                        ),
-                                )
-                                .map((productBlockInstanceValue, index) => (
-                                    <WfoProductBlockKeyValueRow
-                                        fieldValue={productBlockInstanceValue}
-                                        key={index}
-                                    />
-                                ))}
+                            {(!isOutsideCurrentSubscription || !hideDetails) &&
```

Comment:
Think it would be nice to replace this with a named predicate. The `!`'s and `||` have quite some cognitive load for the reader

Reply:
By adding a named predicate